ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/DefaultBlockchain.java

```diff
@@ -530,8 +555,15 @@ private void setGenesis(final Block genesisBlock) {
         throw new IllegalStateException("Blockchain is missing genesis block data.");
       }
       if (!genesisHash.get().equals(genesisBlock.getHash())) {
+        final String firstLine;
+        if (dataDirectory != null) {
+          firstLine =
+              "Supplied genesis block does not match stored chain data in " + dataDirectory + "\n";
+        } else {
+          firstLine = "Supplied genesis block does not match stored chain data.\n";
```

Comment:
Even if we do not pass a data directory will still have the implicit one. It would be nice to log the effective dataDirectory in that case as well.

Reply:
YES because that I can never remember

Reply:
Hmm yeah I think dataDirectory will always be not null as long as it's run from an actual command line, it can only be null in certain tests when we run Besu differently and misses the default CLI stuff, I could be wrong though, I'll look into it

Reply:
1368608